### PR TITLE
[jsscripting] Bump openhab-js to 4.0.0 & Clean-Ups

### PR DESCRIPTION
**Pull request by florian-h05**
* Upgrade the included openhab-js library to version 4.0.0
* Remove the internal `Lifecycle` script extension, since openHAB core provides a version with same functionality since openHAB 3.0.0 (see commit message for more details)

/cc @rkoshak

### CODE REVIEW

File: bundles/org.openhab.automation.jsscripting/README.md [hunk 1]
```diff
@@ -291,16 +292,17 @@ Use the `SCRIPT` transformation with JavaScript Scripting by:
...
 The openHAB JavaScript library provides type definitions for most of its APIs to enable code completion is IDEs like [VS Code](https://code.visualstudio.com).
-To use the type definitions, install the [`openhab` npm package](https://npmjs.com/openhab) (read the [installation guide](https://github.com/openhab/openhab-js#custom-installation) for more information).
-If an API does not provide type definitions and therefore autocompletion won‘t work, the documentation will include a note.
+To use the type definitions, install the [`openhab` npm package](https://npmjs.com/openhab) (read the [installation guide](https://github.com/openhab/openhab-js#custom-installation) for more information), and import the used namespaces with `const { rules, triggers, items } = require('openhab');` (adjust this to your needs).
+If an API does not provide type definitions and therefore autocompletion won't work, the documentation will include a note.
 
 ### Items
 
-The items namespace allows interactions with openHAB items.
+The Items namespace allows interactions with openHAB items.
```
Comment (jlaur) [inline bundles/org.openhab.automation.jsscripting/README.md:300]: It seems to be in lowercase below? Perhaps:
```suggestion
The `items` namespace allows interactions with openHAB items.
```

?

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! Only one question.

File: bundles/org.openhab.automation.jsscripting/README.md (hunk 1 above)
Comment (florian-h05) [reply bundles/org.openhab.automation.jsscripting/README.md:300]: Thanks for the catch. I first have to correct this on the openhab-js repo so I don’t revert that the next time I update the addon README.

Comment (florian-h05) [reply bundles/org.openhab.automation.jsscripting/README.md:300]: I will probably add some doc updates to this PR coming from openhab-js.

Comment (florian-h05) [reply bundles/org.openhab.automation.jsscripting/README.md]: Done.

**Comment by florian-h05** [comment]
@jlaur Ready for re-review!

**Review by jlaur — APPROVED** [review]
Thanks!

**Pull request merged**